### PR TITLE
Add limit switches to dream intake.

### PR DESCRIPTION
**Pull request by J-Double-J**

### CODE REVIEW

File: src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java [hunk 1]
```diff
@@ -4,12 +4,15 @@
...
 
+import edu.wpi.first.wpilibj.DigitalInput;
 import edu.wpi.first.wpilibj.Talon;
 
 public class DreamIntakeSubsystem extends BaseSubsystem {
     private Talon dreamIntakeSideBeltTalon;
     private Talon dreamIntakeMainBeltTalon;
     private Talon dreamIntakeLiftTalon;
+    private DigitalInput dreamIntakeUpSwitch;
+    private DigitalInput dreamIntakeDownSwitch;
```
Comment (bvisness) [inline src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java:15]: You forgot to set the value of these variables again. You need to add stuff like this to the constructor: `this.dreamIntakeUpSwitch = wiringConfig.getDreamIntakeUpSwitch();`.

File: src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java [hunk 2]
```diff
@@ -32,12 +35,10 @@ public void setLiftSpeed(double speed) {
     }
 
     public boolean isCompletelyOut() {
-        // TODO Fill in
-        return false;
+        return dreamIntakeDownSwitch.get();
```
Comment (bvisness) [inline src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java:42]: This code looks fine to me except that the methods are named really strangely. We don't describe the dream's position as "out" and "in" so maybe we should rename this methods `isCompletelyDown()` and `isCompletelyUp()`.

Comment (bvisness) [inline src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java:42]: Actually, leave that for now. That's an issue for another time.

Comment (J-Double-J) [inline src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java:42]: Wow just as I was about to fix that.

File: src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java [hunk 3]
```diff
@@ -17,6 +20,8 @@ public DreamIntakeSubsystem(RobotConfig robotConfig) {
         dreamIntakeSideBeltTalon = wiringConfig.getDreamIntakeSideBeltTalon();
         dreamIntakeMainBeltTalon = wiringConfig.getDreamIntakeMainBeltTalon();
         dreamIntakeLiftTalon = wiringConfig.getDreamIntakeLiftTalon();
+        this.dreamIntakeUpSwitch = wiringConfig.getDreamIntakeUpSwitch();
+        this.dreamIntakeDownSwitch = wiringConfig.getDreamIntakeDownSwitch();
```
Comment (bvisness) [inline src/org/usfirst/frc2175/subsystem/intake/DreamIntakeSubsystem.java:26]: For consistency's sake, let's switch all of them to have `this.` before the variable name.

**Pull request merged**